src/test/java/org/mifos/integrationtest/cucumber/GSMATransferDef.java

```diff
@@ -0,0 +1,272 @@
+        requestSpec.header("sec-ch-ua", "\"Google Chrome\";v=\"105\", \"Not)A;Brand\";v=\"8\", \"Chromium\";v=\"105\"");
+        requestSpec.header("user-agent", "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/105.0.0.0 Safari/537.36");
+        requestSpec.header("content-type", "application/json;charset=UTF-8");
+        return requestSpec;
+    }
+
+    static String getAlphaNumericString(int size) {
+        String AlphaNumericString = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+                + "0123456789"
+                + "abcdefghijklmnopqrstuvxyz";
+        StringBuilder sb = new StringBuilder(size);
+        for (int i = 0; i < size; i++) {
+            int index = (int) (AlphaNumericString.length() * Math.random());
+            sb.append(AlphaNumericString.charAt(index));
+        }
+        return sb.toString();
+    }
+
+    protected static String setBodySavingsProduct() {
+        // Generating product name and shortname
+        String name = new StringBuilder().append("product").append(getAlphaNumericString(3)).toString();
+        String shortName = getAlphaNumericString(3);
+
+        SavingsProduct savingsProduct = new SavingsProduct("USD", 2, 1, 4, 1, 365, "1", name, shortName, "1", 5, new ArrayList<>(), new ArrayList<>(), new ArrayList<>(), new ArrayList<>(), "en");
+
+        return savingsProduct.toString();
+    }
+
+    protected static String setBodySavingsApprove() {
+        SavingsApprove savingsApprove = new SavingsApprove(currentDate, "en", "dd MMMM yyyy");
+        return savingsApprove.toString();
+    }
+
+    protected static String setBodySavingsAccount() throws JsonProcessingException {
+        // Getting resourceId and clientId
+        CreatePayerClientResponse createPayerClientResponse = objectMapper.readValue(
+                responsePayerClient, CreatePayerClientResponse.class);
+        SavingsProductResponse savingsProductResponse = objectMapper.readValue(
+                responseSavingsProduct, SavingsProductResponse.class
+        );
+
+        SavingsAccount savingsAccount = new SavingsAccount(savingsProductResponse.getResourceId(), 5, false, false, false, false, false, 1, 4, 1, 365, currentDate, "en", "dd MMMM yyyy", "dd MMM", new ArrayList<>(), createPayerClientResponse.getClientId(), "");
+
+        return savingsAccount.toString();
+    }
+
+    protected static String setBodySavingsActivate() {
+        SavingsActivate savingsActivate = new SavingsActivate(currentDate, "en", "dd MMMM yyyy");
+        return savingsActivate.toString();
+    }
+
+    protected static String setSavingsDepositAccount(int amount) {
+        SavingsAccountDeposit savingsAccountDeposit = new SavingsAccountDeposit("dd MMMM yyyy", "en", 1, amount, currentDate);
+        return savingsAccountDeposit.toString();
+    }
+
+    protected static RequestSpecification setHeadersClient(RequestSpecification requestSpec) {
+        requestSpec.header("fineract-platform-tenantid", tenant);
```

Comment:
use minimal and configurable headers. also reuse the method

Reply:
they have different letter case for headers that's why can'y reuse for loan and savings.

Reply:
Http is not case sensitive. Try reusing